refinery/tool_manager/tests.py

```diff
@@ -1598,12 +1663,236 @@ def test_get_tool_launch_config(self, tool_data_mock):
                     ),
-                    WorkflowTool.GALAXY_TO_REFINERY_MAPPING_LIST: []
+                    WorkflowTool.GALAXY_TO_REFINERY_MAPPING_LIST: [
+                        {
+                            self.tool.GALAXY_DATASET_HISTORY_ID:
+                                self.FAKE_DATASET_HISTORY_ID,
+                            self.tool.REFINERY_FILE_UUID: self.node.file_uuid,
+                            WorkflowTool.ANALYSIS_GROUP: 0
+                        }
+                    ]
                 }
             }
         )
         self.assertTrue(tool_data_mock.called)
 
+    def test__has_dataset_collection_input_true(self):
+        self.create_valid_tool(ToolDefinition.WORKFLOW)
+        with mock.patch.object(
+            WorkflowTool, "_get_workflow_dict",
+            return_value=galaxy_workflow_dict_collection
+        ) as get_workflow_dict_mock:
+            self.assertTrue(self.tool._has_dataset_collection_input())
+            self.assertTrue(get_workflow_dict_mock.called)
+
+    def test__has_dataset_collection_input_false(self):
+        self.create_valid_tool(ToolDefinition.WORKFLOW)
+        self.assertFalse(self.tool._has_dataset_collection_input())
+
+    def test_analysis_group_numbers_list_dsc_collection_workflow(self):
+        has_dataset_collection_input_mock = (
+            self.has_dataset_collection_input_mock_true.start()
+        )
+        self.create_valid_tool(
+            ToolDefinition.WORKFLOW,
+            file_relationships=self.LIST
+        )
+        self.update_nodes()
+        self.tool._create_collection_description()
+        self.assertTrue(has_dataset_collection_input_mock.called)
+
+        for file_mapping in self.tool._get_galaxy_file_mapping_list():
+            self.assertEqual(file_mapping[WorkflowTool.ANALYSIS_GROUP], 0)
+
+    def test_analysis_group_numbers_list_dsc_non_collection_workflow(self):
+        has_dataset_collection_input_mock = (
+            self.has_dataset_collection_input_mock_false.start()
+        )
+        self.create_valid_tool(
+            ToolDefinition.WORKFLOW,
+            file_relationships=self.LIST
+        )
+        self.update_nodes()
+        self.tool._create_collection_description()
+
+        self.assertTrue(has_dataset_collection_input_mock.called)
+        for index, file_mapping in \
+                enumerate(self.tool._get_galaxy_file_mapping_list()):
+            self.assertEqual(file_mapping[WorkflowTool.ANALYSIS_GROUP], index)
+
+    def test_analysis_group_numbers_pair_dsc_collection_workflow(self):
+        has_dataset_collection_input_mock = (
+            self.has_dataset_collection_input_mock_true.start()
+        )
+        self.create_valid_tool(
+            ToolDefinition.WORKFLOW,
+            file_relationships=self.PAIR
+        )
+        self.update_nodes()
+        self.tool._create_collection_description()
+        self.assertTrue(has_dataset_collection_input_mock.called)
```

Comment:
Would it be worth trying to factor out some of the repeated lines between these tests?